cpp/src/stream_compaction/distinct_helpers.hpp

```diff
@@ -42,13 +43,28 @@ auto constexpr reduction_init_value(duplicate_keep_option keep)
   }
 }
 
+template <typename RowHasher>
+using hash_set_type =
+  cuco::static_set<size_type,
+                   size_type,
```

Comment:
```suggestion
                   cuco::extent<int64_t>,
```
Similar to #16053, use the proper extent type to avoid potential overflow with large input.